algorithms/src/merkle_tree/merkle_path.rs

```diff
@@ -125,16 +125,22 @@ impl<P: MerkleParameters> FromBytes for MerklePath<P> {
         //  If you are seeing this message, please be proactive in bringing it up :)
         let parameters = {
             let setup_message_length: u64 = FromBytes::read_le(&mut reader)?;
+            if setup_message_length > 1024 * 1024 * 10 {
```

Comment:
To sanity check, if I were to reduce this from a `u64` to a `u16`, would this mitigate the need for checking here?

(We can safely assume this will not exceed `u16`; it won't exceed `u8` within reason, though we don't know what the future holds for us)

Reply:
I think this question actually applies to most (if not all) of these cases

Reply:
Yes, reducing the size of the length prefix from `u64` to `u16` will completely mitigate the issue for vectors of bytes (and other small objects), and reduce the scale of the issue in the rest of cases, but note that a `u16::MAX` (65535) number of large objects in vectors might also be an issue.